store.go

```diff
@@ -132,3 +132,37 @@ func (this *Storage) getPage(page string) map[string]*Socket {
 	this.pageMu.RLock()
 	return this.memory.getPage(page)
 }
+
+func (this *Storage) SetGroups(sock *Socket) error {
+	this.pageMu.Lock()
```

Comment:
Can you add a new lock to the storage struct? It should use a separate lock for the groups map.